src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -146,15 +179,19 @@ private Optional<ThingAssociations> getThingsAssociatedWithCoreDevice() {
         try {
             Stream<Thing> cloudThings = RetryUtils.runWithRetry(
                     retryConfig, iotAuthClient::getThingsAssociatedWithCoreDevice,
-                            "get-things-associated-with-core-device", logger);
+                    "get-things-associated-with-core-device", logger);
 
             Stream<Thing> localThings = thingRegistry.getAllThings();
-            ThingAssociations associations =  ThingAssociations.create(cloudThings);
+            ThingAssociations associations = ThingAssociations.create(cloudThings);
             associations.setLocalThings(localThings);
             return Optional.of(associations);
+        } catch (AccessDeniedException e) {
+            logger.atInfo().cause(e).log(
```

Comment:
in this case, logging the stacktrace is not every useful, so omit the `cause`.

This should probably be at warn too, not just info.

Verify with testing that we actually get an `AccessDeniedException`, we have found that that exception isn't actually thrown in many cases, and we needed to change our code to catch any sdk exception and then look at the http status code.

Reply:
Good point. I will write a test for that. I think info is fine, we intentionally wanted to make it so that if that permission is not set the refresh is "disabled". I could go with warn though, don't really know